src/main/java/Duke.java

```diff
@@ -1,10 +1,293 @@
+            taskDescription = words[2];
+            if (words.length > 3) {
+                date = words[3];
+            }
+            loadData(taskType, isDone, taskDescription, date);
+        }
+        System.out.println("Loaded Save File");
+    }
+
+    private static void loadData(String command, boolean isDone, String description, String date) {
+        switch(command) {
+        case "todo":
+            tasks.add(new ToDo(description));
+            if (isDone) {
+                tasks.get(tasks.size() - 1).setDone();
+            }
+            break;
+        case "deadline":
+            tasks.add(new Deadline(description, date));
+            if (isDone) {
+                tasks.get(tasks.size() - 1).setDone();
+            }
+            break;
+        case "event":
+            tasks.add(new Event(description, date));
+            if (isDone) {
+                tasks.get(tasks.size() - 1).setDone();
+            }
+            break;
+        default:
+            System.out.println("Cannot load line");
+        }
+    }
+
+    private static void saveData() throws IOException {
+        createSaveDirectory();
+        createSaveFile();
+        FileWriter writer = new FileWriter("data/duke.txt", false);
+        for (Task task : tasks) {
+            writer.write(task.getTaskType() + " | " + task.getStatusIcon() + "| "
+                    + task.getTaskDescription());
+            if (task instanceof Deadline || task instanceof Event) {
+                writer.write(" | " + task.getTime());
+            }
+            writer.write("\n");
+        }
+        writer.close();
+    }
+
+    private static void createSaveDirectory() {
+        File saveDirectory = new File("data");
+        if (!saveDirectory.exists()) {
+            if (!saveDirectory.mkdir()) {
+                System.out.println("Failed to create new directory.");
+            }
+        }
+    }
+
+    private static void createSaveFile() throws IOException {
+        File saveFile = new File("data/duke.txt");
+        if (saveFile.createNewFile()){
+            System.out.println("Save file created.");
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcomeMessage();
+        Scanner in = new Scanner(System.in);
+        try {
+            readSaveData();
+        } catch (FileNotFoundException e) {
+            System.out.println(("Save File not found."));
+        }
+        parseInput(in);
+        try {
+            saveData();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+        System.out.println("Goodbye, see you next time!");
```

Comment:
You might want to combine the two `try-catch` blocks to make the happy path more prominent.